core/tests/protractor_desktop/additionalEditorFeatures.js

```diff
@@ -383,6 +385,390 @@ describe('Full exploration editor', function() {
+        await richTextEditor.appendPlainText('You must be feeling great?');
+      });
+      await explorationEditorPage.saveChanges();
+      await users.logout();
+
+      await users.login('user11@editor.com');
+      await general.openEditor(explorationId, false);
+      await waitFor.visibilityOf(
+        element(by.css('.modal-content')),
+        'Lost Changes Modal taking too long to appear');
+      await explorationEditorPage.discardLostChanges();
+      await explorationEditorMainTab.expectContentToMatch(
+        async function(richTextChecker) {
+          await richTextChecker.readPlainText('You must be feeling great?');
+        }
+      );
+      await users.logout();
+    });
+
+  it(
+    'should show a discard the changes and download the lost changes file',
+    async function() {
+      await users.createUser(
+        'user13@editor.com',
+        'user13Editor');
+      await users.createUser(
+        'user14@editor.com',
+        'user14Editor');
+      await users.login('user13@editor.com');
+      await workflow.createExploration(true);
+      var explorationId = await general.getExplorationIdFromEditor();
+      await explorationEditorMainTab.setStateName('first card');
+      await explorationEditorPage.navigateToSettingsTab();
+      await explorationEditorSettingsTab.setTitle('Testing lost changes modal');
+      await explorationEditorSettingsTab.setCategory('Algebra');
+      await explorationEditorSettingsTab.setObjective('To assess happiness.');
+      await explorationEditorSettingsTab.openAndClosePreviewSummaryTile();
+      await explorationEditorPage.saveChanges();
+      await workflow.addExplorationManager('user14Editor');
+      await explorationEditorPage.navigateToMainTab();
+
+      await explorationEditorMainTab.setContent(async function(richTextEditor) {
+        await richTextEditor.appendPlainText('How are you feeling?');
+      });
+      await action.waitForAutosave();
+      await users.logout();
+
+      await users.login('user14@editor.com');
+      await general.openEditor(explorationId, true);
+      await explorationEditorMainTab.setContent(async function(richTextEditor) {
+        await richTextEditor.appendPlainText('You must be feeling great?');
+      });
+      await explorationEditorPage.saveChanges();
+      await users.logout();
+
+      await users.login('user13@editor.com');
+      await general.openEditor(explorationId, false);
+      await waitFor.visibilityOf(
+        element(by.css('.modal-content')),
+        'Lost Changes Modal taking too long to appear');
+      await explorationEditorPage.discardLostChanges();
+      await explorationEditorMainTab.expectContentToMatch(
+        async function(richTextChecker) {
+          await richTextChecker.readPlainText('You must be feeling great?');
+        }
+      );
+      await users.logout();
+    });
+
+  afterEach(async function() {
+    await general.checkForConsoleErrors([]);
+  });
```

Comment:
Please move this to the bottom of this file.

Reply:
Done.